### DISCUSSION TITLE
AIO-H3-10.0-E unable to read register 41000

### DISCUSSION THREAD

**Discussion by PaRez280** [nathanmarlor/foxess_modbus#566] [Q&A] [answered]
inverter model AIO-H3-10.0-E
Master : 1.18 - Slave : 1.00 - Manager : 2.81
Everything was working until FoxESS - Modbus v1.8.0 and Home Assistant Core Update 2023.12.3.
When attempting any FoxESS update - Modbus or Home Assistant Core entities are in N/A status.
Is the problem in the inverter firmware version?
Or will there be another problem? Thank you.

**Comment by canton7** [comment]
What does the "Connection Status" entity say? You'll need v1.10.1 of the integration to see this.

Alternatively, look at your logs (Settings - System - Logs). There will be error messages from foxess_modbus.

**Reply by PaRez280** [reply]
Logger: homeassistant.helpers.frame
Zdroj: helpers/frame.py:156
První výskyt: 17:06:39 (2 výskyty)
Naposledy logováno: 17:06:41

Detected that custom integration 'foxess_modbus' accesses hass.components.websocket_api. This is deprecated and will stop working in Home Assistant 2024.9, it should be updated to import functions used from websocket_api directly at custom_components/foxess_modbus/services/websocket_api.py, line 16: hass.components.websocket_api.async_register_command(get_charge_periods), please create a bug report at https://github.com/nathanmarlor/foxess_modbus/issues
Detected that custom integration 'hacs' accesses hass.components.frontend. This is deprecated and will stop working in Home Assistant 2024.9, it should be updated to import functions used from frontend directly at custom_components/hacs/frontend.py, line 68: hass.components.frontend.async_register_built_in_panel(, please create a bug report at https://github.com/hacs/integration/issues

**Reply by PaRez280** [reply]
Tato chyba pochází z vlastní integrace.

Logger: custom_components.foxess_modbus.modbus_controller
Zdroj: custom_components/foxess_modbus/modbus_controller.py:315
integrace: FoxESS - Modbus (dokumentace, problémy)
První výskyt: 17:07:05 (1 výskyty)
Naposledy logováno: 17:07:05

tcp://192.168.0.139:502 247 - 5 failed poll attempts: now not connected. Last error: Error reading registers. Type: RegisterType.HOLDING; start: 41000; count: 1; slave: 247 from tcp://192.168.0.139:502: Exception Response(131, 3, IllegalAddress)

**Reply by canton7** [reply]
Did you recently upgrade your inverter firmware?

Can you update to the latest foxess_modbus version, then go to Developer Tools - Services, select "FoxESS - Modbus: Read Registers", then enter:

* Inverter: Select your inverter
* Start address: 41000
* Count: 1
* Type: try both "input" and "holding"

And see what the output is?

@TonyM1958, I dont suppose you're aware of any changes to the H3-AIO?

**Reply by PaRez280** [reply]
no change to H3-AIO was made


error: >-
  Error reading registers. Type: RegisterType.INPUT; start: 41000; count: 1;
  slave: 247 from tcp://192.168.0.139:502: Modbus Error: [Input/Output] Modbus
  Error: [Invalid Message] No response received, expected at least 8 bytes (0
  received)


error: >-
  Error reading registers. Type: RegisterType.HOLDING; start: 41000; count: 1;
  slave: 247 from tcp://192.168.0.139:502: Exception Response(131, 3,
  IllegalAddress)

**Reply by canton7** [reply]
Did you just upgrade foxess_modbus? What version did you update from?

Maybe the H3-AIO doesn't export Work Mode, which would be a shame...

**Reply by PaRez280** [reply]
Yes,upgrade from version 1.8 to 1.10.1
Version 1.8 is the last working one.

**Reply by canton7** [reply]
Can you try the pre-release v1.10.2b3?

**Reply by PaRez280** [reply, marked as answer]
Yes, v1.10.2b3 is working. Great

**Reply by canton7** [reply]
Thanks!

**Reply by PaRez280** [reply]
Unfortunately I have to report a problem with entities in version v1.10.2b3.  Battery Charge still shows 0KW and battery discharge shows the wrong reading
